mesonbuild/wrap/wrap.py

```diff
@@ -218,17 +218,16 @@ def get_file(self):
 
     def get_git(self):
         revno = self.wrap.get('revision')
-        if self.wrap.values.get('clone-recursive', '').lower() == 'true':
-            subprocess.check_call(['git', 'clone', '--recursive', self.wrap.get('url'),
-                                   self.directory], cwd=self.subdir_root)
-        else:
-            subprocess.check_call(['git', 'clone', self.wrap.get('url'),
-                                   self.directory], cwd=self.subdir_root)
+        subprocess.check_call(['git', 'clone', self.wrap.get('url'),
+                                self.directory], cwd=self.subdir_root)
         if revno.lower() != 'head':
             if subprocess.call(['git', 'checkout', revno], cwd=self.dirname) != 0:
                 subprocess.check_call(['git', 'fetch', self.wrap.get('url'), revno], cwd=self.dirname)
-                subprocess.check_call(['git', 'checkout', revno],
-                                      cwd=self.dirname)
+                subprocess.check_call(['git', 'checkout', revno], cwd=self.dirname)
+        if self.wrap.values.get('clone-recursive', '').lower() == 'true':
+            subprocess.check_call(['git', 'submodule', 'update',
+                                    '--init', '--checkout', '--recursive'],
+                                    cwd=self.dirname)
```

Comment:
__[Flake8]__ [E127] continuation line over-indented for visual indent

<sub>posted by Sider</sub>